src/lib/DateInput.svelte

```diff
@@ -11,32 +11,38 @@
 
   const dispatch = createEventDispatcher<{ select: undefined }>()
 
-  const defaultDate = new Date()
+  /** Default date to use before value is assigned */
+  export let defaultDate = new Date();
 
   // inner date value store for preventing value updates (and also
   // text updates as a result) when date is unchanged
-  const innerStore = writable(defaultDate)
+  const innerStore = writable(null);
   const store = (() => {
     return {
       subscribe: innerStore.subscribe,
       set: (d: Date) => {
-        if (d.getTime() !== $innerStore.getTime()) {
+        console.log('*** store set', {d});
+        if (d && (d.getTime() !== $innerStore?.getTime())) {
           innerStore.set(d)
           value = d
+        } else if (!d && $innerStore) {
+          innerStore.set(null);
         }
       },
     }
   })()
 
   /** Date value */
-  export let value = defaultDate
+  export let value = null;
   $: store.set(value)
+
+  console.log('*** value', {value});
   /** The earliest value the user can select */
   export let min = new Date(defaultDate.getFullYear() - 20, 0, 1)
   /** The latest value the user can select */
   export let max = new Date(defaultDate.getFullYear(), 11, 31, 23, 59, 59, 999)
   /** Placeholder text to show when input field is empty */
-  export let placeholder = '2020-12-31 23:00:00'
+  export let placeholder = 'yyyy-MM-dd HH:mm:ss'
```

Comment:
I'd say this should just stay as it was, and when `value` is null, then we can ignore/override the `placeholder` value

Reply:
reverted